### PR TITLE
Changed Requirement of fixed folder with psscriptroot and xaml dynami…

### PR DESCRIPTION
**Pull request by chwilfing**
…c path replacement, hope this make using it easier without fixed path requirement.

### CODE REVIEW

File: MainWindow.xaml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 ﻿<Window
      xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
      xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" x:Name="MainWindow"
-     Title="[TOOL] O365 Data Retriever" Height="928" Width="1547" ResizeMode="CanMinimize" Icon="C:\O365DataRetriever\Images\PS_Logo.png" WindowStartupLocation="CenterScreen" Background="#c6d7f2" Cursor="Hand" FontFamily="Myriad Pro" WindowStyle="ThreeDBorderWindow">
+     Title="[TOOL] O365 Data Retriever" Height="928" Width="1547" ResizeMode="CanMinimize" Icon="{0}\Images\PS_Logo.png" WindowStartupLocation="CenterScreen" Background="#c6d7f2" Cursor="Hand" FontFamily="Myriad Pro" WindowStyle="ThreeDBorderWindow">
```
Comment (VeronicaGeek) [inline MainWindow.xaml]: Works great 👍

File: O365DataRetriever.ps1 [hunk 2]
```diff
@@ -6,7 +6,8 @@ O365 Data Retriever - Release 0.1.0 (Aug 22nd, 2018)
 Add-Type -AssemblyName PresentationFramework, PresentationCore, WindowsBase, system.windows.forms
 
 #Refer to the XAML file
-[xml]$xml = Get-Content "C:\O365DataRetriever\MainWindow.xaml"
+[string]$xamlContent = Get-Content $(Join-Path -Path $PSScriptRoot -ChildPath 'MainWindow.xaml' )
```
Comment (VeronicaGeek) [inline O365DataRetriever.ps1]: Works great 👍

File: O365DataRetriever.ps1 [hunk 3]
```diff
@@ -630,9 +631,11 @@ $ConnectButton.Add_Click( {
 		#Check ExecutionPolicy is set to "Unrestricted" on computer running the tool (for version 0.1.0)
 		$ExecutionPolicy = Get-ExecutionPolicy
 		if($ExecutionPolicy -ne "Unrestricted"){
-			Write-Host "For this release 0.1.0, the execution policy must be set to Unrestricted. Please change it and try again." -ForegroundColor White -BackgroundColor DarkRed
-			$Window.Close()
-			break
+            Write-Host "For this release 0.1.0, the execution policy must be set to Unrestricted. Current Setting $ExecutionPolicy." -ForegroundColor White -BackgroundColor DarkRed
+            Write-Host "We will set it to Unrestricted for this process but not more!" -ForegroundColor Green
```
Comment (VeronicaGeek) [inline O365DataRetriever.ps1]: I don't like the idea of the tool setting the Execution Policy to "unrestricted". I prefer people to set it themselves after reading the pre-reqs and acknowledging it. Or not running the tool if there's security concerns.

File: O365DataRetriever.ps1 [hunk 4]
```diff
@@ -732,7 +735,7 @@ $ConnectButton.Add_Click( {
 
 		#If user is MFA enabled
 		$UserAuthN = Get-MsolUser -UserPrincipalName ($creds).UserName 
-		if(($UserAuthN.StrongAuthenticationMethods) -ne $null){
+		if($UserAuthN.StrongPasswordRequired -eq $true) {
```
Comment (VeronicaGeek) [inline O365DataRetriever.ps1]: OK.

File: O365DataRetriever.ps1 [hunk 5]
```diff
@@ -703,7 +706,7 @@ $ConnectButton.Add_Click( {
             $ConnectButton.Foreground = "White"
             $ConnectButton.Content = "Connected"
             $ConnectButton.IsHitTestVisible = $false
-            $AdminPwdTextbox.Clear()
+            #$AdminPwdTextbox.Clear()
```
Comment (VeronicaGeek) [inline O365DataRetriever.ps1]: Clearing the password box is only not to leave the asterisks on display. It has never impacted the rest of the script. Needs more investigation with VSCode or more issues about this before changing it.

File: O365DataRetriever.ps1 [hunk 6]
```diff
@@ -765,11 +768,13 @@ $ConnectButton.Add_Click( {
         Import-PSSession $SfBOsession -AllowClobber | Out-Null
 
         #Connect to SPO
-		Write-Host "Connecting to SharePoint Online..." -ForegroundColor Cyan
-        Connect-SPOService -Url https://$SPOTenant-admin.sharepoint.com -Credential $creds
+        Write-Host "Connecting to SharePoint Online..." -ForegroundColor Cyan
+        $creds = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList ($AdminUserName, $(ConvertTo-SecureString -String $AdminPwdTextbox.Password -AsPlainText -Force))
```
Comment (VeronicaGeek) [inline O365DataRetriever.ps1]: Needs more investigation with VSCode or more issues related to this before changing it.

File: O365DataRetriever.ps1 [hunk 7]
```diff
@@ -765,11 +768,13 @@ $ConnectButton.Add_Click( {
...
+        Connect-SPOService -Credential $creds -Url "https://$SPOTenant-admin.sharepoint.com"
 
         #Connect to the Compliance Center
-		Write-Host "Connecting to the Compliance Center..." -ForegroundColor Cyan
+        Write-Host "Connecting to the Compliance Center..." -ForegroundColor Cyan
+        $creds = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList ($AdminUserName, $(ConvertTo-SecureString -String $AdminPwdTextbox.Password -AsPlainText -Force))
```
Comment (VeronicaGeek) [inline O365DataRetriever.ps1]: Needs more investigation with VSCode or more issues related to this before changing it.

File: O365DataRetriever.ps1 [hunk 8]
```diff
@@ -941,7 +946,7 @@ $ConnectButton.Add_Click( {
 
                 # Save the file...
                 if ($SaveGAFileDialog.ShowDialog() -eq 'OK') {
-                    (Get-MsolRoleMember -RoleObjectId $GARoleObjectId) | Select-Object DisplayName, EmailAddress, IsLicensed | Export-Csv $($SaveGAFileDialog.filename) -NoTypeInformation -Encoding UTF8
+                    Get-MsolRoleMember -RoleObjectId $GARoleObjectId | Select-Object DisplayName, EmailAddress, IsLicensed | Export-Csv $($SaveGAFileDialog.filename) -NoTypeInformation -Encoding UTF8
```
Comment (VeronicaGeek) [inline O365DataRetriever.ps1]: Unfortunately, this doesn't change the outcome, and issue still persists.

File: O365DataRetriever.ps1 [hunk 9]
```diff
@@ -1625,8 +1630,18 @@ $ConnectButton.Add_Click( {
 
                 # Save the file...
                 if ($SaveAllSCFileDialog.ShowDialog() -eq 'OK') {
-                    Get-SPOSite -Limit All| Select-Object Title, url, StorageLimit, StorageUsed, Owner, LockState, Template, ConditionalAccessPolicy |
-                        Export-Csv $($SaveAllSCFileDialog.filename) -NoTypeInformation -Encoding UTF8
+                    Get-SPOSite -Limit All | `
```
Comment (VeronicaGeek) [inline O365DataRetriever.ps1]: Works great 👍

File: O365DataRetriever.ps1 [hunk 10]
```diff
@@ -1734,8 +1749,17 @@ $ConnectButton.Add_Click( {
 
                 # Save the file...
                 if ($SaveAllPersoSCFileDialog.ShowDialog() -eq 'OK') {
-                    Get-SPOSite -Limit All -IncludePersonalSite $true | Where-Object {$_.Url -like "*/personal*"} | Select-Object Title, Url, StorageLimit, StorageUsed, Owner, SharingCapability, LockState, Template |
-                        Export-Csv $($SaveAllPersoSCFileDialog.filename) -NoTypeInformation -Encoding UTF8
+                    Get-SPOSite -Limit All -IncludePersonalSite $true | `
```
Comment (VeronicaGeek) [inline O365DataRetriever.ps1]: Works great 👍

File: O365DataRetriever.ps1 [hunk 11]
```diff
@@ -1935,6 +1959,4 @@ $DisconnectButton.Add_Click( {
 #endregion
 
 #Show the GUI
-$Window.ShowDialog()
-
-
+$Window.ShowDialog() | Out-Null
```
Comment (VeronicaGeek) [inline O365DataRetriever.ps1]: OK!

File: O365DataRetriever.ps1 [hunk 12]
```diff
@@ -585,7 +586,7 @@ $ExportSkypeUsersButton = $Window.findname('ExportSkypeUsersButton')
 
 
 #Focus on AdminTextBox when Window loads
-$AdminTextBox.Focus()
+$AdminTextBox.Focus() | Out-Null
```
Comment (VeronicaGeek) [inline O365DataRetriever.ps1]: OK

**Comment by chwilfing** [comment]
sorry, but i'm just too stupid to get it right with the existing pull request, i'll start another pull request with the changes requested. sorry for the inconvenience

**Pull request closed without merging**